src/main/java/com/compuware/jenkins/totaltest/TotalTestCTBuilder.java

```diff
@@ -1585,8 +1585,14 @@ public FormValidation doCheckCredentialsId(@QueryParameter final String value)
 		 *            Value passed from the config.jelly "fileExtension" field
 		 * @return validation message
 		 */
-		public FormValidation doCheckReportFolder(@QueryParameter final String value)
+		public FormValidation doCheckReportFolder(@QueryParameter final String value, @AncestorInPath Item project)
 		{
+			if (project == null) {
+				Jenkins.get().checkPermission(Jenkins.ADMINISTER);
+			} else {
+				project.checkPermission(Item.CONFIGURE);
+			}

```

Comment:
Information about the controller file system should be limited to users with Overall/Administer permission.

I recommed either returning `FormValidation#ok` if that permission is missing, possibly with a message telling users that form validation for this field is only available to users with Overall/Administer permission.

The same applies to the other code location.

Reply:
Hello @daniel-beck,

As per you recommendation, I made changes, please review for same. Please let us know your review points, if any

Thanks,
Ajay

Reply:
Hello @daniel-beck,
Appreciate your quick response and help.
Could you please provide more details about the issue or possible solution and it location. 
Please let us know if there are any other issues, so we can work on it.
Thanks,
Ajay

Reply:
Hello @aasat, The information about the path you want to give are only from the controller file system, they are quite sensitive and must be available only to the administrator of your instance, so it is better not to give any information at all.

Here, Daniel suggests you to only look at the `Jenkins.Administer` permission ,which means to remove your check on `Item.Configure` or even on the `project` and to replace it only by the example he provided you.

Reply:
Hello @daniel-beck, @Kevin-CB,
Thanks for your quick response.
As suggested, updated code

Thanks,
Ajay